### PR TITLE
Some minor JS cleanup

### PR DESCRIPTION
**Pull request by askvortsov1**
See commits for details

### CODE REVIEW

File: js/src/common/Translator.ts [hunk 1]
```diff
@@ -4,34 +4,44 @@ import username from './helpers/username';
...
+   * @ts-ignore
+   */
+  protected formatter = new RichMessageFormatter(null, this.formatterTypeHandlers(), mithrilRichHandler);
+
+  setLocale(locale: string) {
+    this.formatter.locale = locale;
+  }
 
-    this.formatter = new RichMessageFormatter(null, this.formatterTypeHandlers(), mithrilRichHandler);
+  addTranslations(translations: { [key: string]: string }) {
```
Comment (davwheat) [inline js/src/common/Translator.ts:23]: We should use the `Record` type for this.

```suggestion
  addTranslations(translations: Record<string, string>) {
```

File: js/src/common/Translator.ts [hunk 2]
```diff
@@ -45,7 +55,7 @@ export default class Translator {
     return parameters;
   }
 
-  trans(id, parameters) {
+  trans(id: string, parameters: any) {
```
Comment (davwheat) [inline js/src/common/Translator.ts:58]: Could we replace `any` with something else? Should `parameters` also be marked as optional using `?:`?

Maybe `parameters?: Record<string, unknown>` would work?

File: js/src/common/Translator.ts [hunk 3]
```diff
@@ -4,34 +4,44 @@ import username from './helpers/username';
...
-
-  preprocessParameters(parameters) {
+  /**
+   * A temporary system to preprocess parameters.
+   * Should not be used by extensions.
+   * TODO: An extender will be added in v1.x.
+   *
+   * @internal
+   */
+  protected preprocessParameters(parameters) {
```
Comment (davwheat) [inline js/src/common/Translator.ts:44]: See below comment about type -- same type should be used up here.

`parameters: Record<string, unknown>`?

File: js/src/common/states/ModalManagerState.js [hunk 4]
```diff
@@ -11,22 +11,11 @@ export default class ModalManagerState {
    * @public
    */
   show(componentClass, attrs) {
-    // Breaking Change Compliance Warning, Remove in Beta 15.
     if (!(componentClass.prototype instanceof Modal)) {
       // This is duplicated so that if the error is caught, an error message still shows up in the debug console.
       console.error('The ModalManager can only show Modals');
       throw new Error('The ModalManager can only show Modals');
```
Comment (davwheat) [inline js/src/common/states/ModalManagerState.js:17]: Saves bytes and we're already editing this area anyway :)

```suggestion
      const invalidModalWarning = 'The ModalManager can only show Modals.';
      console.error(invalidModalWarning);
      throw new Error(invalidModalWarning);
```

File: js/src/common/Translator.ts [hunk 5]
```diff
@@ -4,34 +4,44 @@ import username from './helpers/username';
...
-    this.translations = {};
+  /**
+   * A map of translation keys to their translated values.
+   */
+  translations = {};
+
+  /**
+   * The underlying ICU MessageFormatter util.
+   *
+   * @ts-ignore
```
Comment (davwheat) [inline js/src/common/Translator.ts:15]: What's this ignoring?

**Review by davwheat — CHANGES_REQUESTED** [review]

File: js/src/common/Translator.ts (hunk 2 above)
Comment (askvortsov1) [reply js/src/common/Translator.ts:58]: That makes sense to me. Although then we should default to an empty object.

File: js/src/common/Translator.ts (hunk 5 above)
Comment (askvortsov1) [reply js/src/common/Translator.ts:15]: Hmm, it's not erroring anymore. Weird...

**Review by davwheat — APPROVED** [review]

**Pull request merged**